iocore/eventsystem/ProxyAllocator.cc

```diff
@@ -40,21 +42,22 @@ thread_alloc(Allocator &a, ProxyAllocator &l)
 void
 thread_freeup(Allocator &a, ProxyAllocator &l)
 {
-  void *head   = (void *)l.freelist;
-  void *tail   = (void *)l.freelist;
-  size_t count = 0;
-  while (l.freelist && l.allocated > thread_freelist_low_watermark) {
-    tail       = l.freelist;
-    l.freelist = *(void **)l.freelist;
-    --(l.allocated);
-    ++count;
+  if (!cmd_disable_freelist_more) {
```

Comment:
Instead of this wonky boolean, can we make a freelist bitfield? so, e.g.

```
if (cmd_disable_freelist && cmd_disable_freelist & DISABLE_PROXY_ALLOCATOR_FREELIST) { ...
```

and (for the old option):

```
if (cmd_disable_freelist && cmd_disable_freelist & DISABLE_GLOBAL_FREELIST) { ...
```

Or some such ? Possibly add some "likely" or "unlikely" as well?